content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+
+```
+public class genericList<T>
+
+{
+    
+    void add(T input){}
+}
+```
+
+On the same note, a list can be of integer or string data type. They are declared differently but the syntax is the same as shown:
+```
+using system;
+
+{
+private class genericListExample
+static void Main ()
+{
+    genericlist<int> list1 = new genericlist<int>(); //This declares a  list of type int.
+    
+     genericlist<string> list2 = new genericlist<string>(); //This declares a list of type string.
+     
+      genericlist<int> list1 = new genericlist<int>(); //This declares a list of type int.
+}
+}
+```
+
+A simple c# example of generic list implementation might help in understanding.
+
+```
+using system;
+using system.collection.generic; // A system namespace for implementing generic lists.
+
+namespace GenericListExample
+{
+    public class Generic<T>  // Declare Generics.
+    {
+        public void GenFunction(T printvalue)
+        {
+            Console.WriteLine(printvalue)
+        }
+    }
+    public class program
+    {
+        public static void Main(string [] args)
+        {
+            Console.WriteLine("Printing integer value");
+            Generic<int> gen = new Generic<int>();
+            gen.GenFunction(10);
+            
+            Console.WriteLine("Printing string value");
+            Generic<string> genString = new Generic<string>();
+            gen.GenFunction("Hello world");
+            
+        }
+    }
+}
+```
+
+Note that _<T>_ is a placeholder for data types and replaced by datatype at compile time.
+
+### 2. Hashsets
+This is an implementation of a set. A set is like a big bag of entries with no particular order rather than a list. They are based on the discrete mathematic's idea of a 'set', which is the grouping of entries that can contain one, two, or many distinct items.  Sorting in a hash set is impossible because the entries have to be sorted. Use of a hash set is very much applicable where we have to store a large amount of data that doesn't have to be sorted eg storing the identity of cars passing through an entrance gate. A program example:
+
+```
+ using System;
+using System.Collections.Generic;
+  
+class MyHashSet {
+
+    static public void Main()
+    {
+  
+        // Creating HashSet
+     
+        HashSet<string> myhash1 = new HashSet<string>();
+  
+        // Add the elements in HashSet
+  
+        myhash1.Add("Ndung'u");
+        myhash1.Add("Elijah");
+
+        // Creating another HashSet
+     
+        HashSet<string> myhash2 = new HashSet<string>();
+        myhash2.Add("Section");
+        myhash2.Add("Git");
+   
+  
+        // Using ExceptWith method
+        myhash1.ExceptWith(myhash2);
+        foreach(var ele in myhash1)
+        {
+            Console.WriteLine(ele);
+        }
+    }
+}
+```
+
+We note that many operations available for the list are available for hash sets except that:
+
+* We CANNOT add an object at a specific position since the elements are not in any order.
```

Comment:
> We CANNOT add an object at a specific position since the elements are not in any order.

I would say it’s not necessary to put *CANNOT* in all caps. If you really want to emphasize it I would bold or italicize it.